src/linear_algebra/eigenspace.lean

```diff
@@ -118,35 +119,34 @@ begin
   have h_lin_dep : ¬ linear_independent K (λ n : ℕ, (f ^ n) v),
   { apply not_linear_independent_of_infinite, },
   -- Therefore, there must be a nonzero polynomial `p` such that `p(f) v = 0`.
-  obtain ⟨p, h_eval_p, h_p_ne_0⟩ : ∃ p, eval₂ am f p v = 0 ∧ p ≠ 0,
-  { simp only [not_imp.symm],
-    exact not_forall.1 (λ h, h_lin_dep ((linear_independent_powers_iff_eval₂ f v).2 h)) },
-  -- Then `p(f)` is not invertible.
-  have h_eval_p_not_unit : eval₂_ring_hom' am _ f p ∉ is_unit.submonoid (End K V),
+  obtain ⟨p, ⟨h_mon, h_eval_p⟩⟩ : is_integral K f
+    := is_integral_of_noetherian (by apply_instance) f,
```

Comment:
```suggestion
  obtain ⟨p, ⟨h_mon, h_eval_p⟩⟩ : is_integral K f :=
    is_integral_of_noetherian (by apply_instance) f,
```

Reply:
I see later on that you added a lemma for `is_integral`, would it be possible to move that to before this line so you can apply it directly? (I didn't check extremely well that the type/dependencies are OK.)